inst/templates/ad_adcoeq.R

```diff
@@ -0,0 +1,257 @@
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  )
+
+# Add analysis parameter variables
+adcoeq <- adcoeq %>%
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT
+  )
+
+# Derive Date/Time and Analysis Day ----
+# See the "Derive/Impute Numeric Date/Time and Analysis Day" vignette section
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#datetime)
+
+# Add questionnaire analysis date (ADT) and treatment start date (TRTSDT)
+adcoeq <- adcoeq %>%
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+# Derive visit info ----
+# See the "Visit and Period Variables" vignette for more information:
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis visit (AVISIT, AVISITN)
+adcoeq <- adcoeq %>%
+  mutate(
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      str_detect(VISIT, "SCREEN") ~ "SCREENING",
+      ADT <= TRTSDT ~ "Baseline",
+      TRUE ~ str_to_title(VISIT)
+    ),
+    AVISITN = case_when(
+      AVISIT == "Baseline" ~ 0,
+      str_detect(AVISIT, "SCREEN") ~ -1,
+      str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
+      TRUE ~ NA_integer_
+    )
+  )
+
+# Derive results ----
+# See the "Derive Results (AVAL, AVALC)" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#aval)
+
+# Derive analysis result (AVAL, AVALC)
+adcoeq <- adcoeq %>%
+  mutate(
+    # Invert the original scores as they indicate anxiety and CoEQ item 6
+    # is used in calculating the subscale for "Positive Mood"
+    AVAL = if_else(PARAMCD == "COEQ06", 100 - QSSTRESN, QSSTRESN),
+    AVALC = if_else(PARAMCD == "COEQ20", QSORRES, NA_character_)
+  )
+
+
+# Derive summary records ----
+
+# See the "Example 2(Deriving a Summary Record)" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#example-2-deriving-a-summary-record)
+
+# For the Control of Eating Questionnaire, four subscales are derived.
+# These subscales are derived as the sum or the average across a subset of the
+# various items/questions.
+
+# The subscales are defined as follows:
+
+# 1) Craving Control: Calculate average of items 9, 10, 11, 12 and 19.
+
+# 2) Craving for Sweet: Calculate average of items 3, 13, 14 and 15.
+
+# 3) Craving for Savoury: Calculate average of items 4, 16, 17 and 18.
+
+# 4) Positive Mood: Calculate average of items 5, 7, 8 and 6 (reversed).
+
+# Derive four subscales
+adcoeq <- adcoeq %>%
+  call_derivation(
+    derivation = derive_summary_records,
+    variable_params = list(
+      params(
+        filter_add = PARAMCD %in% c(
+          "COEQ09", "COEQ10", "COEQ11", "COEQ12", "COEQ19"
+        ) & AVISITN >= 0,
```

Comment:
```suggestion
        filter_add = PARAMCD %in% c("COEQ09", "COEQ10", "COEQ11", "COEQ12", "COEQ19" ) ,
```
why don't we calculate subscale scores for screening records?

Reply:
I will again need a confirmation here from @kathrinflunkert  and @Sonali-netizen  as in what is followed

Reply:
Most likely the values from screening are not used for analysis, we can leave it as it is

Reply:
Agree